workflow-event-listener/src/main/java/com/netflix/conductor/contribs/listener/archive/ArchivingWorkflowToS3.java

```diff
@@ -0,0 +1,71 @@
+package com.netflix.conductor.contribs.listener.archive;
+
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.netflix.conductor.core.dal.ExecutionDAOFacade;
+import com.netflix.conductor.core.exception.TerminateWorkflowException;
+import com.netflix.conductor.core.listener.WorkflowStatusListener;
+import com.netflix.conductor.metrics.Monitors;
+import com.netflix.conductor.model.WorkflowModel;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ArchivingWorkflowToS3 implements WorkflowStatusListener {
+
+    private static final Logger LOGGER =
+            LoggerFactory.getLogger(ArchivingWorkflowToS3.class);
+    private final ExecutionDAOFacade executionDAOFacade;
+
+    private final ArchivingWorkflowListenerProperties properties;
+
+    private final AmazonS3 s3Client;
+
+    private final String bucketName;
+    private final String bucketRegion;
+    private final ObjectMapper objectMapper;
+
+    public ArchivingWorkflowToS3(ExecutionDAOFacade executionDAOFacade, ArchivingWorkflowListenerProperties properties) {
+        this.executionDAOFacade = executionDAOFacade;
+        this.properties = properties;
+        bucketName = properties.getWorkflowS3ArchivalDefaultBucketName();
+        bucketRegion = properties.getWorkflowS3ArchivalBucketRegion();
+        s3Client = AmazonS3ClientBuilder.standard().withRegion(bucketRegion).build();
+        objectMapper = new ObjectMapper();
+    }
+
+    @Override
+    public void onWorkflowCompleted(WorkflowModel workflow) {
+        archiveWorkflow(workflow);
+    }
+
+    @Override
+    public void onWorkflowTerminated(WorkflowModel workflow) {
+        archiveWorkflow(workflow);
+    }
+
+    private void archiveWorkflow(final WorkflowModel workflow) {
+        // Only archive unsuccessful workflows if enabled
+        if (!properties.getWorkflowArchiveUnsuccessfulOnly() || !workflow.getStatus().isSuccessful()) {
+            final String fileName = workflow.getWorkflowId() + ".json";
+            final String filePathPrefix = workflow.getWorkflowName();
+            final String fullFilePath = filePathPrefix + '/' + fileName;
+
+            try {
+                // Upload workflow as a json file to s3
+                s3Client.putObject(bucketName, fullFilePath, objectMapper.writeValueAsString(workflow));
+                LOGGER.info(
```

Comment:
Can this result in a large number of log messages?

Reply:
Yeah, but these are in line with their changes. They are printing info logs for each workflows getting archived.

Reply:
Their implementation I agree. But on Clari's scale, we will start printing millions of info logs. Let's make this debug.

Reply:
done